scripts/irods/test/test_prep_genquery_iterator.py

```diff
@@ -982,3 +982,88 @@ def main(rule_args, callback, rei):
             f"[iteration=1] row: ['{self.admin.session_collection}']",
             f"[iteration=2] row: ['{self.admin.session_collection}']"
         ])
+
+    @unittest.skipUnless(plugin_name == 'irods_rule_engine_plugin-python', 'Requires PREP.')
+    def test_genquery_iterator_supports_genquery2_order_by_clause__issue_7909(self):
+        iterations = 5
+
+        # Create some data objects and build a single string which captures the expected order
+        # of the rows returned by GenQuery2.
+        data_name_prefix = 'test_genquery_iterator_supports_genquery2_order_by_clause__issue_7909'
+        expected_output = ''
+        for i in reversed(range(iterations)):
+            self.admin.assert_icommand(['itouch', f'{data_name_prefix}.{i}'])
+            expected_output += f"['{data_name_prefix}.{i}']\n"
+
+        # Create a rule file which, when executed, lists the data objects in the admin's
+        # session collection in reverse order.
+        rule_file = f'{self.admin.local_session_dir}/test_genquery_iterator_supports_genquery2_order_by_clause__issue_7909.r'
+        with open(rule_file, 'w') as rf:
+            rf.write(dedent(f'''
+            def main(rule_args, callback, rei):
+                from genquery import Query, Parser
+                for r in Query(callback, 'DATA_NAME', "COLL_NAME = '{self.admin.session_collection}'", order_by='DATA_NAME desc', parser=Parser.GENQUERY2):
+                    callback.writeLine('stdout', f'{{r}}')
+            INPUT null
+            OUTPUT ruleExecOut
+            '''))
+
+        # Execute the rule.
+        rep_instance = IrodsConfig().default_rule_engine_plugin + '-instance'
+        self.admin.assert_icommand(['irule', '-r', rep_instance, '-F', rule_file], 'STDOUT', expected_output)
```

Comment:
Does this actually care about the order of the items in `expected_output`?

Reply:
Yes. Do you feel you've spotted a bug?

Reply:
I wasn't sure if `assert_icommand` was just checking for each string in the list in the stdout contents, or if it is actually checking to see that they appear in order. Maybe worth reversing/scrambling the order of `expected_output` and making sure that the assertion here fails?

Reply:
The `expected_output` is a single string containing the list of data object names in reversed order. Will double check.

Reply:
Correct. I've confirmed `expected_output` is working as expected. Printing the string and removing the call to `reversed()` causes the test to fail.